apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManagerExtendService.java

```diff
@@ -84,4 +92,10 @@ public AbstractTracerContext createTraceContext(String operationName, boolean fo
     public void statusChanged(final GRPCChannelStatus status) {
         this.status = status;
     }
+
+    void handleIgnoreSuffixPatternsChanged() {
+        if (StringUtil.isNotBlank(ignoreSuffixPatternsWatcher.getIgnoreSuffixPatterns())) {
+            ignoreSuffixArray = ignoreSuffixPatternsWatcher.getIgnoreSuffixPatterns().split(",");
```

Comment:
@kezhenxu94 @zifeihan I want to discuss about this detail, including the previous implementation of `TraceIgnoreExtendService`.
The `ignoreSuffixArray` doesn't have `volatile` key word, due to another thread is watching and updating the value, the latest value will not be affected in some threads immediately.
According to this kind of update is not urgent, it should be fine. But I am to check with you two, have we agreed this is our conclusion?

Reply:
I agree with you, since this is not an urgent operation, I think it should be fine. What do you think? @kezhenxu94

Reply:
> The ignoreSuffixArray doesn't have volatile key word, due to another thread is watching and updating the value, the latest value will not be affected in some threads immediately. According to this kind of update is not urgent, it should be fine.

Since most of the notifications of config centers are not real-time (have delay) themselves, the fields not being `volatile` may be acceptable if it has no side effects.

> According to this kind of update is not urgent, it should be fine. But I am to check with you two, have we agreed this is our conclusion?

I don't remember we have this conclusion before. TBH, I didn't notice this part indeed 🤦🏻‍♂️

Reply:
This isn't a conclusion yet, but I noticed this, 1st dynamic configuration is sampling rate, which is an AtomicReference, so it is fine. Then, ignore plugin, it has been approved and merged, with no `volatile`. And I can see @zifeihan approved this too.

So, I want to ask, what do you think about this? Sometimes, no `volatile` is tricky about when it effects in the runtime.

Reply:
Once we have this agreement, I want this to be noticed in a new GitHub issue and mail list for searching in the future.

Reply:
I prefer to add `volatile` or similar semantics (like `AtomicXxxx`)

Reply:
> I prefer to add `volatile` or similar semantics (like `AtomicXxxx`)

Agree with you.

Reply:
@zifeihan OK, then you need to fix your previous merged PR ASAP, we are closing the release.